lib/Doctrine/Common/Lexer/AbstractLexer.php

```diff
@@ -258,6 +258,11 @@ protected function scan($input)
         $flags = PREG_SPLIT_NO_EMPTY | PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_OFFSET_CAPTURE;
         $matches = preg_split($regex, $input, -1, $flags);
 
+        if (false === $matches) {
+            // Work around https://bugs.php.net/78122
+            $matches = array(array($input, 0));
```

Comment:
Should we trigger a deprecation telling that input that triggers error described by `preg_last_error` will not be supported (as in, we throw an exception) in the next major version of the lexer? Or are we supposed to always allow this situation?

Reply:
I don't think that's necessary: there was talk about replacing `doctrine/lexer` with HOA parser in `doctrine/annotations` and `doctrine/dbal`. We can always add the deprecation in 1.1 if we decide to do a 2.0 release.

Reply:
Oh ok I didn't know this package was going to be abandoned. I still think a deprecation should be thrown as a way to tell the user they are probably misusing that method and that bad things may happen. That would mean a deprecation w/o any hint about a 2.0 version of the package:
```suggestion
            @trigger_error(sprintf(
                'input "%s" caused preg_split() to fail with error code %d (description at https://www.php.net/manual/en/pcre.constants.php). "%s" may not behave as expected',
                $input,
                preg_last_error(),
                __METHOD__
            ), E_USER_DEPRECATED);
            $matches = array(array($input, 0));
```